spack-ext/lib/jcsda-emc/spack-stack/stack/templates/mpi.lua

```diff
@@ -32,11 +32,19 @@ setenv("MPI_F77", "@MPIF77@")
 setenv("MPI_F90", "@MPIF90@")
 
 -- intel specific mpi wrapper environment variables
-setenv("I_MPI_CC",  os.getenv("CC"))
-setenv("I_MPI_CXX", os.getenv("CXX"))
-setenv("I_MPI_F77", os.getenv("F77"))
-setenv("I_MPI_F90", os.getenv("FC"))
-setenv("I_MPI_FC",  os.getenv("FC"))
+local i_mpi_cc = os.getenv("CC")
+local i_mpi_cxx = os.getenv("CXX")
+local i_mpi_f77 = os.getenv("F77") or ""
+local i_mpi_f90 = os.getenv("FC")
+local i_mpi_fc = os.getenv("FC")
+
+if i_mpi_cc and i_mpi_cxx and i_mpi_f90 and i_mpi_fc then
```

Comment:
Is there an instance where these aren't all getting set? Isn't all or nothing based on compilers.lua? I might not under the problem exactly, so if you could add steps to reproduce in the issue description that would be appreciated.